tracing-journald/src/lib.rs

```diff
@@ -109,6 +114,48 @@ impl Subscriber {
         self.field_prefix = x;
         self
     }
+
+    #[cfg(not(unix))]
+    fn send_payload(&self, _opayload: &[u8]) -> io::Result<()> {
+        Err(io::Error::new(
+            io::ErrorKind::Unsupported,
```

Comment:
MSRV is currently set to 1.49 but `io::ErrorKind::Unsupported` was stabilized in 1.53.

This wasn't caught on CI because it's gated behind `#[cfg(not(unix))]` 🙃

Reply:
ughhh, that's annoying. currently we don't build MSRV on windows (or macOS) to avoid having a really huge number of CI builds... but maybe we should be.

Reply:
Replacing it with `NotFound` would be fine. We shouldn't panic.

Reply:
Hmm, #2232 is a breaking change --- if anyone is currently compiling this code across multiple operating systems, the method will suddenly disappear for them. I would be fine with making the entire impl `#[cfg(unix)]` in a future breaking change release, but I think we should fix the MSRV violation without breaking the API in a point release first. Therefore, I would really prefer to maintain the current behavior of returning an error on unsupported operating systems.

If we don't like `NotFound`, we could just use `io::ErrorKind::Other`, which should exist on our MSRV.

Reply:
> And by the way, did anyone actually hit this MSRV violation in their build?

I "hit" it, but only inasmuch as I was doing a minver check on the repository from windows.

For #2231 I just replaced this with `ErrorKind::Other`.

> MSRV tests on other OSes

I think in general it's fine to assume the Linux build will catch things, as the OS-gated functionality is rare enough, and we can fix violations when they come up.

If we *do* want to check other targets, it should be via cross-compile where possible since linux hosted runner time is cheaper, and just a check is sufficient.

(Ultimately CI-wise this comes back to the downside of a monorepo being testing the entire monorepo to verify a subpackage-local change... though this is sometimes desirable since e.g. tracing-core gets a significant amount of test coverage only from testing of tracing or even tracing-subscriber...)

> don't compile on windows

There are AIUI four reasonable choices (of which only the latter pair are legally semver compatible):

- `#![cfg]` the entire crate to only expose any API on platforms where it makes sense. This is IIRC the approach taken by winapi. Consumers `#[cfg]` their usage.
- `compile_error!` on platforms where the crate doesn't make any sense. Consumers need to use a target-dependent dependency as well as `#[cfg]` their usage.
- `panic!` (or error) entrypoints into the API on unsupported platforms. Consumers `#[cfg]` their usage (or tolerate an error on supported platforms as well).
- Stub functionality on unsupported platforms. IIUC tracing-journald is purely a sink, so stubbing it as `> /dev/null` (`$null`<sub>pwsh</sub>) isn't *wrong*, just perhaps unexpected. Consumers can unconditionally use the API and just implicitly degrade on unsupported platforms.

Reply:
> I think NotFound would be wrong, semantically; after all, we're not searching for anything here

Semantically, we are searching for `JOURNALD_PATH`. This is the error that will be returned when running on a Unix without journald, and unconditionally when running on Windows.